project/project.go

```diff
@@ -66,14 +67,30 @@ func NewProject(context *Context, runtime RuntimeProject, parseOptions *config.P
 
 	if context.EnvironmentLookup == nil {
 		cwd, err := os.Getwd()
+		var isPresent bool
```

Comment:
You should be able to read `ComposeFiles` from context rather than looking through `os.Args`. You can use the directory of the first Compose file.